src/redux/greetingSlice.js

```diff
@@ -0,0 +1,45 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const URL_API = 'http://localhost:3000/api/v1/greetings/random';
+export const fetchRandomGreeting = createAsyncThunk('greetings/fetchRandomGreeting', async () => {
```

Comment:
- [ ] There is an issue rendering the data, I figured that there is a core which is a permission issue, the weird aspect is that I could see the data when I go to the url directly, but I think you need to set the CORS Headers on the Server, ensure that your Rails API server is configured to include the necessary CORS headers in its responses. You can use the rack-cors gem in your Rails application to handle CORS.
![image](https://github.com/mariegrace31/hello-rails-front-end/assets/86200356/4a31e0a5-4a4a-4eb9-98ea-4681787e0cda)

So it would look something like:
```
# config/application.rb

config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins '*' # Update this to your allowed origins
    resource '*', headers: :any, methods: [:get, :post, :options], credentials: true
  end
end
```